ananse/network.py

```diff
@@ -213,408 +180,117 @@ def distance_weight(self, include_promoter=False, include_enhancer=True, alpha=1
-        prom_table = prom_table.reset_index()
-        prom_table["source_target"] = (
-            prom_table["factor"].map(str) + "_" + prom_table["gene"].map(str)
+        weight1 = pd.DataFrame(
+            {
+                "weight": [promoter_weight for z in range(0, remove + 1)],
+                "dist": range(0, remove + 1),
+            }
         )
 
-        f_table = ddf.merge(p, left_on="enhancer", right_on="loc")
-        sum_enh = f_table.groupby(["factor", "gene"])[["binding"]].count()
-        f_table["sum_weighted_logodds"] = (
-            f_table["binding"]
-            .div(f_table["binding"].mean())
-            .apply(np.log, meta=("binding", np.float64))
-            .rmul(50000)
-            .div(f_table["dist"])
+        weight2 = pd.DataFrame(
+            {
+                "weight": [enhancer_weight for z in range(remove + 1, keep1 + 1)],
+                "dist": range(remove + 1, keep1 + 1),
+            }
         )
-        f_table["sum_logodds"] = (
-            f_table["binding"]
-            .div(f_table["binding"].mean())
-            .apply(np.log, meta=("binding", np.float64))
+
+        weight3 = pd.DataFrame(
+            {
+                "weight": [
+                    enhancer_weight
+                    * 2.0
+                    * math.exp(-u * math.fabs(z) / 1e5)
+                    / (1.0 + math.exp(-u * math.fabs(z) / 1e5))
+                    for z in range(1, padding - keep1 + 1)
+                ],
+                "dist": range(keep1 + 1, padding + 1),
+            }
         )
 
-        weight = dd.read_csv(weight)
-        f_table = f_table.merge(weight, how="left", on="dist")
-        f_table["sum_dist_weight"] = f_table["binding"] * f_table["weight"]
+        weight = pd.concat([weight1, weight2, weight3])
+        return weight
+
+    def aggregate_binding(
+        self,
+        binding_fname,
+        tfs=None,
+        up=1e5,
+        down=1e5,
+        alpha=None,
+        promoter=2000,
+        full_weight_region=5000,
+        combine_function="sum",
+    ):
+
+        if combine_function not in ["mean", "max", "sum"]:
+            raise ValueError(
+                "Unknown combine function, valid options are: mean, max, sum"
+            )
 
-        f_table_sum = f_table.groupby(["factor", "gene"]).sum()[
-            ["sum_weighted_logodds", "sum_logodds", "binding", "sum_dist_weight"]
-        ]
-        f_table_max = f_table.groupby(["factor", "gene"])[
-            ["binding", "sum_dist_weight"]
-        ].max()
-
-        f_table_sum = f_table_sum.rename(columns={"binding": "sum_binding"})
-        f_table_max = f_table_max.rename(columns={"binding": "max_binding"})
-        f_table_max = f_table_max.rename(
-            columns={"sum_dist_weight": "max_sum_dist_weight"}
+        padding = max(up, down)
+        if alpha is None:
+            alpha = padding / 10
+
+        if promoter > padding:
+            raise ValueError("promoter region is > region to use")
```

Comment:
I have no idea what this means, and why this is important. Could you add docstrings to this function?